File: Android/DayCareTeacher/DayCareTeacher/app/src/main/java/com/daycare/daycareteacher/model/StudentMedication.kt
package com.daycare.daycareteacher.model

import com.google.gson.annotations.SerializedName

class StudentMedication {
    @SerializedName("studentMedicationID")
     val studentMedicationID: Int? = null
    @SerializedName("studentID")
     val studentID: Int? = null
    @SerializedName("medicationName")
     val medicationName: String? = null
    @SerializedName("agencyID")
     val agencyID: Int? = null
    @SerializedName("units")
     val units: String? = null
    @SerializedName("strength")
     val strength: String? = null
    @SerializedName("doses")
     val doses: String? = null
    @SerializedName("howTaken")
     val howTaken: Any? = null
    @SerializedName("otherMedication")
     val otherMedication: Any? = null
    @SerializedName("startDate")
     val startDate: String? = null
    @SerializedName("endDate")
     val endDate: String? = null
    @SerializedName("isActive")
     val isActive: Boolean? = null
    @SerializedName("isDeleted")
     val isDeleted: Boolean? = null
    @SerializedName("deletedBy")
     val deletedBy: Int? = null
    @SerializedName("deletedDate")
     val deletedDate: String? = null
    @SerializedName("createdBy")
     val createdBy: Int? = null
    @SerializedName("createdDate")
     val createdDate: String? = null
    @SerializedName("updatedDate")
     val updatedDate: String? = null
    @SerializedName("updatedBy")
     val updatedBy: Int? = null
}

File: Web/MS-DayCare_backendLatest/DayCare.Entity/Student/StudentMedication.cs

using JsonApiDotNetCore.Models;
using JsonApiDotNetCore.Services;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace DayCare.Entity.Student
{

    public class StudentMedication : BaseEntity, IHasMeta
    {
        [Attr("StudentMedicationID")]
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Column("StudentMedicationID")]
        public override long Id { get; set; }

        [Required]
        [ForeignKey("Student")]
        [Attr("StudentID")]
        public long StudentID { get; set; }
        
        [StringLength(500)]
        [Attr("StudentMedicationName")]
        public string MedicationName { get; set; }

        [Required]
        [ForeignKey("Agency")]
        [Attr("AgencyID")]
        public long AgencyID { get; set; }
        
        [Attr("MedicationUnits")]
        public long Units { get; set; }

        [StringLength(50)]
        [Attr("MedicationStrength")]
        public string strength { get; set; }


        [Required]
        [ForeignKey("DoseRepeat")]
        [Attr("DoseRepeatID")]
        public long DoseRepeatID { get; set; }

        [Required]
        [ForeignKey("DosageQuantity")]
        [Attr("DosageQuantityID")]
        public long DosageQuantityID { get; set; }

        [Attr("HowTaken")]
        [StringLength(500)]
        public string HowTaken { get; set; }

        [Attr("OtherMedication")]
        [StringLength(500)]
        public string OtherMedication { get; set; }

        [Attr("MedicationStartDate")]
        public DateTime StartDate { get; set; }

        [Attr("MedicationEndDate")]
        public DateTime EndDate { get; set; }

        [Attr("isTeacherAcknowledge")]
        public bool isTeacherAcknowledge { get; set; }

        [Attr("isParentAcknowledge")]
        public bool isParentAcknowledge { get; set; }
        public Dictionary<string, object> GetMeta(IJsonApiContext context)
        {
            try
            {
                return new Dictionary<string, object> {
                { "total-pages",  context.PageManager.TotalPages },
                { "page-size",  context.PageManager.PageSize },
                { "current-page",  context.PageManager.CurrentPage },
                { "default-page-size",  context.PageManager.DefaultPageSize },
            };
            }
            catch (Exception)
            {
                context.PageManager.PageSize = 10;
                return new Dictionary<string, object> {
                { "total-pages",  context.PageManager.TotalPages },
                { "page-size",  context.PageManager.PageSize },
                { "current-page",  context.PageManager.CurrentPage },
                { "default-page-size",  context.PageManager.DefaultPageSize },
            };
            }
        }
    }
}

